test/test__formUI_status_test.py

```diff
@@ -39,6 +41,50 @@ def add_two_widgets(self):
         form.addWidget(QtWidgets.QLabel('test label'), 'Label: ', 'label')
         form.addWidget(QtWidgets.QCheckBox('test checkbox'), 'CheckBox: ', 'checkBox')
 
+    def set_state(self, ii):
+        # define two states for every widget
+        state = [0, 0]
+        state[0] = {
+            'label_value': 'Test label state 0', 'checkbox_value': False, 'combobox_value': 0,
+            'doubleSpinBox_value': 10.0, 'spinBox_value': 10, 'slider_value': 10,
+            'uislider_value': 10, 'radio_value': False, 'textEdit_value': 'test edit 0',
+            'plainTextEdit_value': 'test plain 0', 'lineEdit_value': 'test line 0',
+            'pushButton_value': False}
+        state[1] = {
+            'label_value': 'Test label state 1', 'checkbox_value': True, 'combobox_value': 1,
+            'doubleSpinBox_value': 1.0, 'spinBox_value': 1, 'slider_value': 1, 'uislider_value': 1,
+            'radio_value': True, 'textEdit_value': 'test edit 1',
+            'plainTextEdit_value': 'test plain 1', 'lineEdit_value': 'test line 1',
+            'pushButton_value': True}
+        # set the states
+        # QLabel
+        self.form.getWidget('label').setText(state[ii]['label_value'])
+        # QCheckBox
+        self.form.getWidget('checkBox').setChecked(state[ii]['checkbox_value'])
+        # QComboBox
+        combobox_list = ['test', 'test2']
+        self.form.getWidget('comboBox').addItems(combobox_list)
+        self.form.getWidget('comboBox').setCurrentIndex(state[ii]['combobox_value'])
+        # QDoubleSpinBox
+        self.form.getWidget('doubleSpinBox').setValue(state[ii]['doubleSpinBox_value'])
+        # QSpinBox
+        self.form.getWidget('spinBox').setValue(state[ii]['spinBox_value'])
+        # QSlider
+        self.form.getWidget('slider').setValue(state[ii]['slider_value'])
+        # UISlider
+        self.form.getWidget('uiSliderWidget').setValue(state[ii]['uislider_value'])
+        # QRadioButton
+        self.form.getWidget('radioButton').setChecked(state[ii]['radio_value'])
+        # QTextEdit
+        self.form.getWidget('textEdit').setText(state[ii]['textEdit_value'])
```

Comment:
In order to use the `applyWidgetStates` method I need to change the format of `exampleState`. Then, I would apply some values to the widgets. Do you think this approach is better? It could be more complicated but happy to change it. Should we write an `applyValuesToWidgets` method?

Reply:
After discussions with @lauramurgatroyd  we decided to open a new issue on the default-states dictionary to improve the way they are saved and keep it separate from this PR.